pulsar-client-api/src/main/java/org/apache/pulsar/client/api/Message.java

```diff
@@ -245,4 +245,22 @@
      * @since 2.8.0
      */
     void release();
+
+    /**
+     * Get broker publish time from broker entry metadata.
+     * Note that only if the feature is enabled in the broker then the value is available.
+     *
+     * @since 2.9.0
+     * @return broker publish time from broker entry metadata, or empty if the feature is not enabled in the broker.
+     */
+    Optional<Long> getBrokerPublishTime();
```

Comment:
It's better to add a method for checking if has broker publish time, such as` boolean hasBrokerPublishTime()`;

Reply:
Done